sdk/cosmosdb/cosmos/src/diagnostics/Diagnostics.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { defaultLogger } from "../common/logger";
+
+const _startTime = new Date().getTime();
+const _defaultHeader: DiagnosticHeader = {
+  diagnostics: "Cosmos Diagnostics Summary",
+  diagnosticStartTime: new Date().toLocaleString(),
+  durationInMs: new Date().getTime() - _startTime,
+  activityId: "",
+  data: ["Summary"],
+  requestCharge: "0",
+  transportRequestTimeline: {},
+  contactedRegions: "",
+};
+const _diagnosticHeader: DiagnosticHeader = _defaultHeader;
+
+/**
+ * @internal
+ */
+export function setDiagnostics(message: string = ""): void {
+  const _patternMatch = new RegExp(`[sS]tatusCode: [2-3]\d\d `);
+  if (!_patternMatch.test(message)) {
+    if (message || message !== undefined) {
+      if (_diagnosticHeader || _diagnosticHeader !== undefined) {
```

Comment:
can this `_diagnosticHeader` ever be null? If so, we should have this check everywhere. If not, we should remove it from here as well.